misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -127,45 +122,44 @@ public ModuleCoverage generateReport(JarResolver jarResolver) throws IOException
         }
     }
 
-    private IBundleCoverage analyzeStructure() throws IOException {
+    private CoverageBuilder analyzeStructure() throws IOException {
         final CoverageBuilder coverageBuilder = new CoverageBuilder();
         final Analyzer analyzer = new Analyzer(execFileLoader.getExecutionDataStore(), coverageBuilder);
         analyzer.analyzeAll(classesDirectory.toFile());
-        return coverageBuilder.getBundle(title);
+        return coverageBuilder;
     }
 
-    private void createXMLReport(IBundleCoverage bundleCoverage) throws IOException {
-            XMLFormatter xmlFormatter = new XMLFormatter();
-            File reportFile = new File(target.getReportPath().resolve(
-                    this.module.moduleName().toString()).resolve(REPORT_XML_FILE).toString());
-            reportFile.getParentFile().mkdirs();
-
-            try (FileOutputStream fileOutputStream = new FileOutputStream(reportFile)) {
-                IReportVisitor visitor = xmlFormatter.createVisitor(fileOutputStream);
-                visitor.visitInfo(execFileLoader.getSessionInfoStore().getInfos(),
-                        execFileLoader.getExecutionDataStore().getContents());
-
-                visitor.visitBundle(bundleCoverage, null);
+    private IBundleCoverage getPartialCoverageModifiedBundle(CoverageBuilder coverageBuilder) {
+        return new BundleCoverageImpl(title, coverageBuilder.getClasses(),
+                modifySourceFiles(coverageBuilder.getSourceFiles()));
+    }
 
-                visitor.visitEnd();
-            }
+    private void createXMLReport(IBundleCoverage bundleCoverage) throws IOException {
+        final XMLFormatter xmlFormatter = new XMLFormatter();
+        final File reportFile = new File(target.getReportPath().resolve(
+                this.module.moduleName().toString()).resolve(REPORT_XML_FILE).toString());
+        reportFile.getParentFile().mkdirs();
+        try (FileOutputStream fileOutputStream = new FileOutputStream(reportFile)) {
+            IReportVisitor visitor = xmlFormatter.createVisitor(fileOutputStream);
+            visitor.visitInfo(execFileLoader.getSessionInfoStore().getInfos(),
+                    execFileLoader.getExecutionDataStore().getContents());
+            visitor.visitBundle(bundleCoverage, null);
```

Comment:
why we are passing in a null value here?

Reply:
is this an existing code, it shows as Anjalees change because whitespace changes.

Reply:
@Kishanthan it is the sourcefile locator, we dont need to pass it to xml since it is not used. https://github.com/jacoco/jacoco/blob/master/org.jacoco.report/src/org/jacoco/report/xml/XMLFormatter.java#L70